sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/LiteralExpressionSuite.scala

```diff
@@ -239,6 +239,11 @@ class LiteralExpressionSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvaluation(Literal.create('\n'), "\n")
   }
 
+  test("SPARK-33390: Make Literal support char array") {
+    checkEvaluation(Literal(Array('h', 'e', 'l', 'l', 'o')), "hello")
```

Comment:
Could you add tests in `CatalystTypeConvertersSuite` and `DatasetSuite`, too, according to https://github.com/apache/spark/commit/54fcaafb094e299f21c18370fddb4a727c88d875 ?

Reply:
Thanks for the remind ! Actually I have not seen that commit.